quinn/src/endpoint.rs

```diff
@@ -257,12 +258,19 @@ where
     fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
         let mut recvd = 0;
         loop {
-            match self.socket.poll_recv(cx, &mut self.recv_buf) {
-                Poll::Ready(Ok((n, addr, ecn))) => {
-                    match self
-                        .inner
-                        .handle(now, addr, ecn, (&self.recv_buf[0..n]).into())
-                    {
+            let mut meta = [RecvMeta::default()];
+            match self
+                .socket
+                .poll_recv(cx, &mut [IoSliceMut::new(&mut self.recv_buf)], &mut meta)
+            {
+                Poll::Ready(Ok(msgs)) => {
+                    debug_assert_eq!(msgs, 1);
+                    match self.inner.handle(
+                        now,
+                        meta[0].addr,
+                        meta[0].ecn,
+                        (&self.recv_buf[0..meta[0].len]).into(),
```

Comment:
If we name this last argument, is that enough to get us within the line limit?

Reply:
Yep!